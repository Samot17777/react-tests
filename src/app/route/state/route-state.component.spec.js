import React from 'react';
import {render, shallow, mount} from 'enzyme';
import {RouteStateComponent} from "./route-state.component";

describe('Route state component', () => {
    it('renders correctly', () => {
        const wrapper = render(<RouteStateComponent />);

        expect(wrapper).toMatchSnapshot();
    });

    it('changes state correctly', () => {

        const wrapper = shallow(<RouteStateComponent />);

        expect(wrapper.find('span').text()).toBe("0");
        wrapper.setState({
                exampleNumber: 2
        });

        expect(wrapper.find('span').text()).toBe("2");

        wrapper.find('button').simulate('click');
        wrapper.find('button').simulate('click');

        expect(wrapper.find('span').text()).toBe("4");
    });
});
