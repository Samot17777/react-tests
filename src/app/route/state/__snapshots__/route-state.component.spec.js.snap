// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Route state component renders correctly 1`] = `
<div>
  <span>
    0
  </span>
  <button>
    change
  </button>
</div>
`;
